src/marqo/tensor_search/add_docs.py

```diff
@@ -64,119 +82,185 @@ def threaded_download_and_preprocess_images(
+                                                                                     timeout_ms=int(
+                                                                                         TIMEOUT_SECONDS * 1000),
+                                                                                     metrics_obj=metric_obj)
+                        except PIL.UnidentifiedImageError as e:
+                            media_repo[doc[field]] = e
+                            metric_obj.increment_counter(f"{doc.get(field, '')}.UnidentifiedImageError")
+                            continue
+                        # preprocess image to tensor
+                        if preprocessors is not None and preprocessors['image'] is not None:
+                            if not device or not isinstance(device, str):
+                                raise ValueError("Device must be provided for preprocessing images")
+                            try:
+                                media_repo[doc[field]] = preprocessors['image'](media_repo[doc[field]]).to(device)
+                            except OSError as e:
+                                if "image file is truncated" in str(e):
+                                    media_repo[doc[field]] = e
+                                    metric_obj.increment_counter(f"{doc.get(field, '')}.OSError")
+                                    continue
+                                else:
+                                    raise e
+
+                    elif (modality in [Modality.VIDEO,
+                                       Modality.AUDIO] and marqo_index.type == IndexType.Unstructured) or (
+                            marqo_index.type == IndexType.Structured and media_field_types_mapping[field] in 
+                            [FieldType.AudioPointer, FieldType.VideoPointer]):
+                        if marqo_index.model.properties.get('type') not in [
+                            ModelType.LanguageBind] and modality not in marqo_index.model.properties.get(
+                            'supported_modalities'):
+                            media_repo[doc[field]] = UnsupportedModalityError(
+                                f"Model {marqo_index.model.name} does not support {modality}")
+                            continue
+                        try:
+                            processed_chunks = download_and_chunk_media(doc[field], device, download_headers, modality,
+                                                                        marqo_index, preprocessors)
+                            media_repo[doc[field]] = processed_chunks
+                        except (ffmpeg.Error, S2InferenceError) as e:
+                            logger.error(f"Error processing {modality} file: {str(e)}")
+                            media_repo[doc[field]] = S2InferenceError(f"Error processing {modality} file: {str(e)}")
+
+                    else:
+                        pass
```

Comment:
Should we raise an error here?

Reply:
done

Reply:
All good. A key error (which goes to a 500) is not a bad idea in this case.